proposals/3392/dataset_partitioning.md

```diff
@@ -0,0 +1,301 @@
+### Input filtering dataset facet
+
+inputFacets should be used to express dataset facets related to a particular run. 
+A condition for partition reading is just a special case example of all possible dataset filtering. 
+This document is not trying to solve a full problem of describing all the possible dataset filters, 
+however a facet introduced needs to be extended easily.
+
+```json
+{
+   "$schema": "https://json-schema.org/draft/2020-12/schema",
+   "$id": "https://openlineage.io/spec/facets/1-0-2/FilterInputDatasetFacet.json",
+   "$defs": {
+      "BaseFilterCondition": {},
+      "BinaryFilterCondition": {
+         "allOf": [
+            {
+               "$ref": "BaseFilterCondition"
+            },
+            {
+               "type": "object",
+               "properties": {
+                  "left": {
+                     "type": "BaseFilterCondition"
+                  },
+                  "right": {
+                     "type": "BaseFilterCondition"
+                  },
+                  "operator": {
+                     "enum": [
+                        "OR",
+                        "AND"
+                     ]
+                  }
+               }
+            }
+         ]
+      },
+      "LiteralComparisonCondition": {
+         "allOf": [
+            {
+               "$ref": "BaseFilterCondition"
+            },
+            {
+               "type": "object",
+               "properties": {
+                  "field": {
+                     "type": "String"
+                  },
+                  "value": {
+                     "type": "String"
+                  },
+                  "comparison": {
+                     "enum": [
+                        "EQUAL",
+                        "GREATER_THAN",
+                        "GREATER_EQUAL_THAN",
+                        "LESS_THAN",
+                        "LESS_EQUAL_THAN"
+                     ]
+                  }
+               }
+            }
+         ]
+      },
+      "FilterInputDatasetFacet": {
+         "allOf": [
+            {
+               "$ref": "https://...OpenLineage.json#/$defs/InputDatasetFacet"
+            },
+            {
+               "type": "object",
+               "properties": {
+                  "condition": {
+                     "type": "#/$defs/BaseFilterCondition"
+                  }
+               }
+            }
+         ]
+      }
+   },
+   "type": "object",
+   "properties": {
+      "filter": {
+         "$ref": "#/$defs/FilterInputDatasetFacet"
+      }
+   }
+}
+
+```
+
+
+
+An example filter facet for SQL `SELECT * FROM table WHERE col1 = 7 and col2 = 9` would be:
+
+```json
+{
+  "filter": {
+    "$id": "https://openlineage.io/spec/facets/1-0-0/BinaryFilterCondition.json",
+    "left": {
+      "field": "col1",
+      "value": "7",
+      "comparison": "EQUAL"
+    },
+    "right": {
+      "field": "col2",
+      "value": "9",
+      "comparison": "EQUAL"
+    },
+    "operator": "AND"
```

Comment:
If there are multiple `AND`s, how this should be represented?

Reply:
`AND` is binary condition but `left` and `right` can be composed of other binary conditions as well.

Reply:
For me, this looks like a part of another proposal leaked to partitioning spec. Is that really needed to describe a list of partitions read by Spark or other integration?

Reply:
Could we use the same facet for input & output partitions? And move this facet with filters to another proposal.

Reply:
I agree with @dolfinus - the proposal makes sense even without filtering, and it's a special case that we might want to think about separately, with more clear use case.

Reply:
I missed that, sry.  I will remove this part onto another PR.

Reply:
https://github.com/OpenLineage/OpenLineage/pull/3470